### ISSUE TITLE
Auto filling uses wrong values if rows or columns has been moved via DnD

### ISSUE THREAD

**Issue by olosegres** [handsontable/handsontable#9646]
### Description
Hello again from Actonic!
Auto filling (spread cell values up or down by stretching selected cell) uses wrong values if rows has been moved via DnD.

### Steps to reproduce

1. Drag and drop any row or column
2. Select dropped cell
3. Pull down the corner of the cell (stretch it)

### Actual result
Cells below filled with the value that were on the place of selected cell before DnD

### Expected result
Cells below filled with the actual, visible value of cell that we are spreading

### Demo
https://codesandbox.io/s/wizardly-faraday-m3hbe2
[Screencast from 2022-07-12 15:44:46.webm](https://user-images.githubusercontent.com/3730589/178473510-09228c2f-9304-4e59-bcb2-31b6f943fd9f.webm)

### Your environment
React wrapper version: 12.1
Handsontable version: 12.1
Browser name and version: Chrome 103, Firefox
Operating system: macos, linux

**Comment by AMBudnik** [comment]
Thank you for sharing the bug report. I replicated the issue.

![issue](https://user-images.githubusercontent.com/10757813/178494148-c5560b5c-a115-4f41-b4b1-28474860684c.gif)

The cause of the issue is _formulas_. We have a similar case here https://github.com/handsontable/handsontable/issues/8242 

I will notify you as soon as we fix the issue.

**Comment by AMBudnik** [comment]
Unfortunately it happens even when you just define the plugin. If you comment out the

```js
   formulas: {
     engine: HyperFormula,
     sheetName: `Sheet-1`
   },
```

from your `staticHandsontableConfig` the issue is no longer replicable.

**Comment by olosegres** [comment]
got it, but we can't disable the formulas.
Please, if possible, raise the priority of bugs

**Comment by warpech** [comment]
Might be related with https://github.com/handsontable/handsontable/issues/8513

**Comment by AMBudnik** [comment]
Hi @olosegres 

Is the issue replicable using v 14.0.0? I couldn't open your old demo (I got some errors) but created a new one here https://jsfiddle.net/1ph2q6gy/ and when I move the last row up and autofill it works well. 

Could you please check if that also work for you?

**Comment by olosegres** [comment]
@AMBudnik Looks like it is fixed in 14.0.0 🥳

**Comment by AMBudnik** [comment]
Great! Thank you for your prompt feedback. We can then close this issue as solved.
